backend/src/ee/routes/v1/external-kms-router.ts

```diff
@@ -286,4 +291,89 @@ export const registerExternalKmsRouter = async (server: FastifyZodProvider) => {
       return { externalKms };
     }
   });
+
+  server.route({
+    method: "POST",
+    url: "/fetch-gcp-keys/credential",
+    config: {
+      rateLimit: writeLimit
+    },
+    schema: {
+      body: z.object({
+        region: z.string().trim().min(1),
+        credential: ExternalKmsGcpCredentialSchema
+      }),
+
+      response: {
+        200: z.object({
+          keys: z.array(z.string())
+        })
+      }
+    },
+    onRequest: verifyAuth([AuthMode.JWT, AuthMode.IDENTITY_ACCESS_TOKEN]),
+    handler: async (req) => {
+      const { region, credential } = req.body;
+
+      const results = await server.services.externalKms.fetchGcpKeys({
+        credential,
+        gcpRegion: region,
+        keyName: ""
+      });
+      if (!results) {
+        throw new Error("Failed to validate GCP credential");
+      }
+
+      return results;
+    }
+  });
```

Comment:
best if we can somehow unify the paths for provider-specific endpoints. like it would be better if the path for this was something like `/gcp/keys`

Reply:
why is there a credential suffix in the url?

Reply:
let's also avoid having verbs in the url. let's try to stick to RESTful conventions when possible